src/models/StackedEnsemble/base/neural/tabnet_model_outliers.py

```
@@ -0,0 +1,1909 @@
+        )
+
+        # Add fit_weights used to metrics dict for logging
+        metrics["fit_weights_used"] = fit_weights_value
+        # Add GPU memory usage to metrics if available
+        if torch.cuda.is_available():
+            metrics["peak_gpu_memory_mb"] = peak_mem
+
+        # Store the scaler with the model for future use
+        model._scaler = scaler
+
+        return model, metrics
```

Ruff (0.13.3)

Consider moving this statement to an `else` block

(TRY300)